src/Features/Core/Portable/GenerateConstructorFromMembers/AbstractGenerateConstructorFromMembersCodeRefactoringProvider.cs

```diff
@@ -47,60 +49,117 @@ protected AbstractGenerateConstructorFromMembersCodeRefactoringProvider() : this
+                (action, applicableToSpan) => context.RegisterRefactoring(action, applicableToSpan),
+                (actions) => context.RegisterRefactorings(actions), context.CancellationToken);
+        }
+
+        public async Task<Solution?> ComputeActionForIntentAsync(Document documentBeforeIntent, TextSpan selection, Document currentDocument, string? serializedIntentData, CancellationToken cancellationToken)
+        {
+            using var builder = ArrayBuilder<CodeAction>.GetInstance(out var actions);
+            await ComputeRefactoringsAsync(documentBeforeIntent, selection, (singleAction, applicableToSpan) => actions.Add(singleAction), (multipleActions) => actions.AddRange(multipleActions), cancellationToken).ConfigureAwait(false);
+
+            if (actions.IsEmpty())
+            {
+                return null;
+            }
+
+            // The refactorings returned will be in the following order (if available)
+            // FieldDelegatingCodeAction, ConstructorDelegatingCodeAction, GenerateConstructorWithDialogCodeAction
+            // For now since we don't have an idea of which one to pick, we'll take the action that appears first in the group.
+            var codeAction = actions.First();
+            var operations = await GetCodeActionOperationsAsync(codeAction, cancellationToken).ConfigureAwait(false);
+
+            // Generate ctor will only return an ApplyChangesOperation or potentially document navigation actions.
+            // We can only return edits, so we only care about the ApplyChangesOperation.
+            var applyChangesOperation = operations.SingleOrDefault(operation => operation is ApplyChangesOperation) as ApplyChangesOperation;
+            if (applyChangesOperation == null)
+            {
+                return null;
+            }
+
+            return applyChangesOperation.ChangedSolution;
+
+            static async Task<ImmutableArray<CodeActionOperation>> GetCodeActionOperationsAsync(CodeAction action, CancellationToken cancellationToken)
+            {
+                if (action is CodeActionWithOptions)
+                {
+                    // Generate ctor only returns GenerateConstructorWithDialogCodeAction for CodeActionWithOptions
+                    var dialogAction = (GenerateConstructorWithDialogCodeAction)action;
+
+                    // Usually applying this code action pops up a dialog allowing the user to choose which options.
+                    // We can't do that here, so instead we just take the defaults until we have more intent data.
+                    var options = new PickMembersResult(dialogAction.ViableMembers, dialogAction.PickMembersOptions);
+                    var operations = await dialogAction.GetOperationsAsync(options: options, cancellationToken).ConfigureAwait(false);
+                    return operations == null ? ImmutableArray<CodeActionOperation>.Empty : operations.ToImmutableArray();
```

Comment:
something is very smelly here.  we should discuss this later.

my take on this is: the caller should specify if they even care about optiosn or not.  for Intents, we woudl not, so we wouldnt' have to get them and unpack them.

Reply:
In one of the next versions we can see if we can provide these based on opened files till the refactoring detection. Or if you can scan/see for the common usage in the solution. It is good to what was done in the result so we can report over the chooses we made.